### DISCUSSION TITLE
VOL Order rejection on Bag order (Future Spread order)

### DISCUSSION THREAD

**Discussion by championbarca** [erdewit/ib_insync#579] [Q&A] [answered]
Hi, 
When I am sending BAG Order with combo legs of VIX future (June, Sept). Orders are getting rejected with "Error validating request.-'bS' : cause - VOL order requires non-negative floating point value for volatility.  Do not set it for other order types." I am using latest version ib_insync(0.9.82) with TWS 10.22. Server version returned is 177. I think it's due to API changes because same order works with TWS native API. I have tried with building from source still the same.
I have checked following solution and did see ib_insync is sending correct values to TWS, but may be message format has changed.
https://stackoverflow.com/questions/49971796/ib-api-order-placement-error-cannot-set-vol-attribute-on-non-vol-order

And this same code of ib_insync works with 981 version of TWS. Please suggest resolution.

**Comment by erdewit** [comment, marked as answer]
I've currently no idea what the problem could be. The volatility and volatilityType of the order should be sent as empty fields, at least that's what the IBAPI is doing.

TWS 10.22 seems to have a lot of issues, perhaps try 10.19 stable.

**Reply by championbarca** [reply]
Do you know how to get 10.19 version of TWS ?

**Comment by erdewit** [comment]
It's the stable version on IB's website.

**Comment by championbarca** [comment]
Thanks it worked. Also I see that ib_insync internally fetches nextValidId, but it seems I am getting it as always 1 from IB, do you know why is that so?

**Comment by erdewit** [comment]
1 is the starting id, it can be higher dependent on history.

**Comment by championbarca** [comment]
Reason I am investigating is because I was using ib_insync logic for order_id, but I started getting rejections as Duplicate Order Id. So how that can happen where I get rejection due to duplicate order id and at the same time get NextValidId as 1 from IB. is there a bug or am I missing something ?

**Comment by championbarca** [comment]
I found the reason, Order is rejected with duplicate Order Id but actual reason for order rejection is somewhere before, that's the part of confusion. Thanks for your help. Appreciated.
